code/modules/ruins/objects_and_mobs/necropolis_gate.dm

```diff
@@ -236,17 +240,70 @@ GLOBAL_DATUM(necropolis_gate, /obj/structure/necropolis_gate/legion_gate)
 	resistance_flags = INDESTRUCTIBLE | LAVA_PROOF | FIRE_PROOF | UNACIDABLE | ACID_PROOF
 	var/tile_key = "pristine_tile"
 	var/tile_random_sprite_max = 24
+	var/fall_on_cross = STABLE //If the tile has some sort of effect when crossed
+	var/fallen = FALSE //If the tile is unusable
+	var/falling = FALSE //If the tile is falling
 
 /obj/structure/stone_tile/Initialize(mapload)
 	. = ..()
 	icon_state = "[tile_key][rand(1, tile_random_sprite_max)]"
 
 /obj/structure/stone_tile/Destroy(force)
-	if(force)
+	if(force || fallen)
 		. = ..()
 	else
 		return QDEL_HINT_LETMELIVE
 
+/obj/structure/stone_tile/Crossed(atom/movable/AM)
+	var/turf/T = get_turf(src)
+	if(!istype(T, /turf/open/floor/plating/lava) && !istype(T, /turf/open/chasm)) //nothing to sink or fall into
+		return
+	if(falling || fallen)
+		return
+	var/obj/item/I
+	if(istype(AM, /obj/item))
+		I = AM
+	var/mob/living/L = AM
+	if(isliving(AM))
+		L = AM
+	switch(fall_on_cross)
+		if(COLLAPSE_ON_CROSS, DESTROY_ON_CROSS)
+			if((I && I.w_class >= WEIGHT_CLASS_BULKY) || (L && !(L.movement_type & FLYING) && L.mob_size >= MOB_SIZE_HUMAN)) //too heavy! too big! aaah!
+				collapse()
+		if(UNIQUE_EFFECT)
+			crossed_effect(AM)
+
+/obj/structure/stone_tile/proc/collapse()
+	falling = TRUE
+	var/break_that_sucker
+	if(fall_on_cross == DESTROY_ON_CROSS)
+		break_that_sucker = TRUE
+	playsound(src, 'sound/effects/pressureplate.ogg', 50, TRUE)
+	for(var/i in 1 to 5)
+		animate(src, pixel_x = pixel_x + rand(-1, 1), pixel_y = pixel_y = rand(-1, 1), time = 1)
+		sleep(1)
+	if(break_that_sucker)
+		playsound(src, 'sound/effects/break_stone.ogg', 50, TRUE)
+	else
+		playsound(src, 'sound/mecha/mechmove04.ogg', 50, TRUE)
+	animate(src, alpha = 0, pixel_y = pixel_y - 3, time = 5)
+	fallen = TRUE
+	if(break_that_sucker)
+		QDEL_IN(src, 10)
```

Comment:
you should use a clienttime timer for this if it's to match the animation. QDEL_IN uses game ticks

In fact make a QDEL_IN_CLIENT_TIME next to QDEL_IN please
`#define QDEL_IN_CLIENT_TIME(X, Y) addtimer(CALLBACK(GLOBAL_PROC, /proc/qdel, X), T, TIMER_CLIENT_TIME)`